nptdms/tdms.py

```diff
@@ -225,6 +225,46 @@ def as_dataframe(self, time_index=False, absolute_time=False):
                 temp.append((key, pd.Series(data=value.data, index=index)))
         return pd.DataFrame.from_items(temp)
 
+    def as_hdf(self, filepath):
+        import h5py
+        """
+        Converts the TDMS file into an HDF5 file
+
+        :param filepath: The path of the HDF5 file you want to write to.
+        """
+        # Groups in TDMS are mapped to the first level of the HDF5 hierarchy
+
+        # Channels in TDMS are then mapped to the second level of the HDF5
+        # hierarchy, under the appropriate groups.
+
+        # Properties in TDMS are mapped to attributes in HDF5.
+        # These all exist under the appropriate, channel group etc.
+
+        h5file = h5py.File(filepath, 'w')
+
+        # First right the properties at the root level
+        for property_name, property_value in self.object().properties.items():
+            h5file['/'].attrs[property_name] = property_value
+
+        # Now iterate through groups and channels,
+        # writing the properties and data
+        for group_name in self.groups():
+
+            group = self.object(group_name)
```

Comment:
One thing to be careful of is that there may be a group in the file without an associated group object, eg. it's possible to have a channel at `/'group_name'/'channel_name'` but no group object for `/'group_name'`. Please check for a `KeyError` when getting the group object. And above when you get the root object you should also check for a `KeyError`. You can see an example of something similar in https://github.com/adamreeve/npTDMS/blob/master/nptdms/tdmsinfo.py.